src/commands/oas2tson.js

```diff
@@ -0,0 +1,149 @@
+import { Command } from 'commander'
+import filenamify from 'filenamify'
+import fs from 'fs-extra'
+import _get from 'lodash.get'
+import _trimStart from 'lodash.trimstart'
+import path from 'path'
+import pino from 'pino'
+import { exit } from 'process'
+import YAML from 'yaml'
+import $RefParser from '@apidevtools/json-schema-ref-parser'
+
+import { fromSchema } from '../utils/openapi-schema-to-json-schema-wrapper.cjs'
+import prettier from 'prettier'
+
+const COMPONENT_REF_REGEXP = /#\/components\/schemas\/[^"]+/g
+let outputSchemasMetaData = []
+
+export const adaptSchema = (generatedSchema, name, filename) => {
+  delete generatedSchema.$schema
+  generatedSchema.title = name
+  generatedSchema.$id = `${filename}.json`
+
+  if (generatedSchema.format?.includes('date')) {
+    generatedSchema.tsType = 'Date'
+  }
+}
+
+const processSchema = (schema, schemasPath, definitionKeyword, isArray) => {
+  Object.entries(schema).forEach(([key, value]) => {
+    // for elements in an array the name would be its index if we were
+    // to just use its key, so go into the parsed schema and get the
+    // actual name so the files are more easily identifiable
+    const name = isArray ? value.name : key
+    const filename = _trimStart(filenamify(name, { replacement: '-' }), '-')
+
+    adaptSchema(value, name, filename)
+
+    let schemaAsString = JSON.stringify(value, null, 2)
+    // N.B. - this obviously only supports refs where the string contains 'components/schemas'
+    // if we want to support refs in places other than this, we'll need to revisit this
+    // approach to be more flexible
+    const refs = schemaAsString.match(COMPONENT_REF_REGEXP)
+    refs?.forEach(ref => {
+      let refName = ref.split('/').slice(-1)
+      schemaAsString = schemaAsString.replace(ref, `${refName}.json`)
+    })
+
+    const destinationDir = path.join(schemasPath, 'tempjson')
```

Comment:
why do have to temporarly store files in the user's directory?
IMO we should use a temp directory provided by node

Reply:
Addressed the feedback, used node js os's temporary directory for intermediate operations.

Reply:
Where are you using it? I still see the `path.join` with `tempjson` here

Reply:
From line no. 113, I am passing os.tmpdir() to the method processSchema. And actually I am creating a 'tempjson' folder inside that tempdir, latter on deleting it. (Since even tempdir of nodejs isn't periodically cleared, it's cleared during system boot. Hence creating a inner directory so that we can delete it after usage)